src/rust/ide/view/src/project.rs

```diff
@@ -56,9 +57,30 @@ impl Model {
         Self{app,logger,display_object,graph_editor,searcher,code_editor}
     }
 
-    fn set_style(&self, is_light:bool) {
-        if is_light { self.app.themes.set_enabled(&["dark"])  }
-        else        { self.app.themes.set_enabled(&["light"]) }
+    /// Sets style of IDE.
+    pub fn set_style_internal(&self, is_light:bool) {
+        let root_elem = web::get_element_by_id("root");
+        if is_light {
+            ensogl_theme::builtin::dark::enable(&self.app);
+            match root_elem {
+                Ok(v)  => v.set_class_name("dark-theme"),
+                Err(_) => self.logger.warning("Failed to find root!"),
+            }
+        } else {
+            ensogl_theme::builtin::light::enable(&self.app);
+            match root_elem {
+                Ok(v)  => v.set_class_name("light-theme"),
+                Err(_) => self.logger.warning("Failed to find root!"),
+            }
+        }
+    }
+
+    /// Sets style of IDE by theme name.
+    pub fn set_style(&self, theme:&'static str) {
+        match theme {
+            stringify!(ensogl_theme::builtin::dark) => self.set_style_internal(true),
+            &_                                      => self.set_style_internal(false),
+        }
```

Comment:
This function is error prone and confusing. Why when calling `set_style("foo_bar_baz")` the theme will be switched to light one ? Please, use the solution I provided in my previous review instead.